tests/rptest/tests/scaling_up_test.py

```diff
@@ -627,3 +629,62 @@ def disk_usage_correct(nodes, node_id):
         assert self.consumer.consumer_status.validator.invalid_reads == 0, \
         f"Invalid reads in topic: {topic.name}, invalid reads count: "
         "{self.consumer.consumer_status.validator.invalid_reads}"
+
+    @skip_debug_mode
+    @cluster(num_nodes=6)
+    def test_scaling_up_with_recovered_topic(self):
+        log_segment_size = 2 * 1024 * 1024
+        segments_per_partition = 40
+        msg_size = 256 * 1024
+        partition_count = 10
+        total_records = int(
+            (segments_per_partition * partition_count * log_segment_size) /
+            msg_size)
+
+        si_settings = SISettings(test_context=self.test_context,
+                                 log_segment_size=log_segment_size,
+                                 retention_local_strict=True,
+                                 fast_uploads=True)
+
+        self.redpanda.set_si_settings(si_settings)
+        #start 3 node cluster
+        self.redpanda.start(nodes=self.redpanda.nodes[0:3])
+        # create test topic
+        cli = KafkaCliTools(self.redpanda)
+        topic = TopicSpec(name="recovery-topic",
+                          replication_factor=3,
+                          partition_count=partition_count)
+        rpk = RpkTool(self.redpanda)
+        rpk.create_topic(topic.name,
+                         partition_count,
+                         3,
+                         config={
+                             'redpanda.remote.write': 'true',
+                             'redpanda.remote.read': 'true',
+                             'redpanda.remote.delete': 'false'
+                         })
+        # produce some data
+        cli.produce(topic.name, total_records, msg_size)
+        self.redpanda.wait_for_manifest_uploads()
+
+        total_replicas = 3 * partition_count
+        rpk.delete_topic(topic.name)
+
+        rpk.create_topic(topic.name,
+                         partition_count,
+                         3,
+                         config={
+                             'redpanda.remote.recovery': 'true',
+                             'redpanda.remote.write': 'true',
+                             'redpanda.remote.read': 'true',
+                             'redpanda.remote.delete': 'true'
+                         })
+
+        cli.produce(topic.name, total_records, msg_size)
+
+        self.redpanda.start_node(self.redpanda.nodes[3])
+        self.redpanda.start_node(self.redpanda.nodes[4])
+        self.wait_for_partitions_rebalanced(total_replicas=total_replicas,
+                                            timeout_sec=self.rebalance_timeout)
+        self.redpanda.wait_for_manifest_uploads()
```

Comment:
should we wait for this before adding nodes? (well, I guess both cases are interesting)